source/index.js

```diff
@@ -95,41 +102,65 @@ var create = function(options, callback) {
 
 		pump(json, response);
 
-		(function loop() {
-			var key = keys.pop();
-
-			if(!key) {
-				json.writeObject('_error', error);
-				return json.end();
-			}
-
-			var messages = createMessages(request, query[key]);
+		// Exit early if there is nothing to fetch.
+		if(keys.length === 0) {
+			return endStream(json, error);
+		}
 
-			var write = function(prevent) {
-				if(prevent) {
-					return loop();
+		// The resource queue processes resource streams sequentially.
+		var resourceQueue = async.queue(function worker(task, callback) {
+			pump(task.resource, json.createObjectStream(task.key), function(err) {
+				if(err) {
+					json.destroy();
+					return callback(err);
 				}
+				if(!(/2\d\d/).test(task.response.statusCode)) {
+					error = true;
+				}
+				callback();
+			});
+		}, 1);
 
-				var resource = fetch(messages.request, messages.response);
-
-				pump(resource, json.createObjectStream(key), function(err) {
-					if(err) {
-						return json.destroy();
-					}
-					if(!(/2\d\d/).test(messages.response.statusCode)) {
-						error = true;
-					}
-
-					loop();
-				});
-
-				app(messages.request, messages.response, function(err) {
-					json.destroy();
+		// Asynchronously fetch the resource for a key and push the resulting
+		// stream into the resource queue.
+		var fetchResource = function(key, callback) {
+			var messages = createMessages(request, query[key]);
+			process.nextTick(function() {
```

Comment:
Nope, you're right, it's not needed! I think it's an artifact from some previous playing I was doing. Removed.